uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/Upgrader.php

```diff
@@ -919,13 +919,140 @@ public function upgrade_1037() {
 
     return TRUE;
   }
+  
+  /**
+   * Sets up option values, custom group and custom field
+   * for case type categorization
+   *
+   * @return bool
+   */
+  public function upgrade_1038() {
+    $optionValues = civicrm_api3('OptionValue', 'get', [
+      'option_group_id' => 'cg_extend_objects',
+      'name' => 'civicrm_case_type'
+    ]);
+    if ($optionValues['count'] == 0) {
+      $params = [
+        'option_group_id' => 'cg_extend_objects',
+        'name' => 'civicrm_case_type',
+        'label' => ts('Case Type'),
+        'value' => 'CaseType',
+      ];
+      $this->createOptionValue($params);
+    }
+    
+    $customGroups = civicrm_api3('CustomGroup', 'get', [
+      'extends' => 'CaseType',
+      'name' => 'case_type_category',
+    ]);
+    if ($customGroups['count'] == 0) {
+      $this->createCustomGroup();
+    }
+    
+    $customFields = civicrm_api3('CustomField', 'get', [
+      'custom_group_id' => 'case_type_category',
+    ]);
+    if ($customFields['count'] == 0) {
+      $optionGroupId = $this->createCustomFieldOptionValues();
```

Comment:
Same thing here. Perhaps it would be better to rename this to something like `createCaseTypeCategoryOptionValues()`